condition.py

```diff
@@ -0,0 +1,26 @@
+import os
+import threading
+
+
+def number_generator(thread_number):
+    global counter
+    while True:
+        condition.acquire()
```

Comment:
context manager has to be used here:
`with condition:`
Otherwise, condition.release() should be called.